### PR TITLE
JBTM-3273 -> Implemented code review comments

### PR DESCRIPTION
**Pull request by mayankkunwar**
This PR is to implement missed [code review comments](https://github.com/jbosstm/narayana/pull/1590#pullrequestreview-393941092) on another PR.

MAIN QA_JTA AS_TESTS

!TOMCAT !RTS !JACOCO !XTS !QA_JTS_JACORB !QA_JTS_JDKORB !QA_JTS_OPENJDKORB !BLACKTIE !PERF !LRA !NO_WIN !DB_TESTS !mysql !db2 !postgres !oracle

### CODE REVIEW

File: ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest.java [hunk 1]
```diff
@@ -28,8 +28,8 @@
 import com.arjuna.ats.internal.jta.recovery.arjunacore.XARecoveryModule;
 import com.arjuna.ats.jta.common.jtaPropertyManager;
 import com.arjuna.ats.internal.jta.transaction.arjunacore.TransactionImple;
-import com.arjuna.ats.jta.logging.RecoveryRequired;
 import com.arjuna.ats.jta.logging.jtaLogger;
+import com.arjuna.ats.jta.recovery.RecoveryRequired;
```
Comment (mmusgrov) [inline ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest.java:32]: Where is this used?

File: ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java [hunk 2]
```diff
@@ -59,14 +59,8 @@
...
 import com.arjuna.ats.jta.common.jtaPropertyManager;
-import com.arjuna.ats.jta.logging.RecoveryRequired;
 import com.arjuna.ats.jta.logging.jtaLogger;
-import com.arjuna.ats.jta.recovery.SerializableXAResourceDeserializer;
-import com.arjuna.ats.jta.recovery.XARecoveryResource;
-import com.arjuna.ats.jta.recovery.XARecoveryResourceManager;
-import com.arjuna.ats.jta.recovery.XAResourceOrphanFilter;
-import com.arjuna.ats.jta.recovery.XAResourceRecovery;
-import com.arjuna.ats.jta.recovery.XAResourceRecoveryHelper;
+import com.arjuna.ats.jta.recovery.*;
```
[the remark this answers is not in the document]
Comment (mmusgrov) [reply ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java:63]: We have a checkstyle plugin which is skipped by default because we have a lot of old files, that we don't want to change, which would fail the compilation. But on new code we do encourage/advise that it should follow our checkstyle rules.

To see if your code follows our style guide you have two options:

1. Enable the checkstyle plugin via the [pom file](https://github.com/jbosstm/narayana/blob/master/pom.xml#L377), or

2. Reference the rule file from your IDE settings (for Intelij that would be Settings -> Other Settings -> Checkstyle) and we use the [wildfly rule set](https://github.com/wildfly/wildfly-checkstyle-config/tree/master/src/main/resources/wildfly-checkstyle).

File: ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest2.java [hunk 3]
```diff
@@ -27,8 +27,8 @@
 import com.arjuna.ats.arjuna.tools.RecoveryMonitor;
 import com.arjuna.ats.internal.jta.recovery.arjunacore.XARecoveryModule;
 import com.arjuna.ats.internal.jta.transaction.arjunacore.TransactionImple;
-import com.arjuna.ats.jta.logging.RecoveryRequired;
 import com.arjuna.ats.jta.logging.jtaLogger;
+import com.arjuna.ats.jta.recovery.RecoveryRequired;
```
Comment (mmusgrov) [inline ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest2.java:31]: Where is this used?

File: ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryScanTest.java [hunk 4]
```diff
@@ -28,8 +28,8 @@
 import com.arjuna.ats.internal.jta.recovery.arjunacore.XARecoveryModule;
 import com.arjuna.ats.internal.jta.transaction.arjunacore.TransactionImple;
 import com.arjuna.ats.jta.common.jtaPropertyManager;
-import com.arjuna.ats.jta.logging.RecoveryRequired;
 import com.arjuna.ats.jta.logging.jtaLogger;
+import com.arjuna.ats.jta.recovery.RecoveryRequired;
```
Comment (mmusgrov) [inline ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryScanTest.java:32]: Where is this used?

File: ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (mayankkunwar) [reply ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java:63]: > We can avoid the new RecoveryRequired class and just add an AtomicBoolean to this class instead.

Okay, I will remove the new RecoveryRequired class and add AtomicBoolean

[the remark this answers is not in the document]
Comment (mayankkunwar) [reply ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java:63]: > We have a checkstyle plugin which is skipped by default because we have a lot of old files, that we don't want to change, which would fail the compilation. But on new code we do encourage/advise that it should follow our checkstyle rules.
> 
> To see if your code follows our style guide you have two options:
> 
> 1. Enable the checkstyle plugin via the [pom file](https://github.com/jbosstm/narayana/blob/master/pom.xml#L377), or
> 2. Reference the rule file from your IDE settings (for Intelij that would be Settings -> Other Settings -> Checkstyle) and we use the [wildfly rule set](https://github.com/wildfly/wildfly-checkstyle-config/tree/master/src/main/resources/wildfly-checkstyle).

Maybe checkstyle changes can be done on separate PR?

File: ArjunaJTA/jta/classes/com/arjuna/ats/jta/recovery/RecoveryRequired.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
+package com.arjuna.ats.jta.recovery;
+
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+public class RecoveryRequired {
+
+    private static ReentrantReadWriteLock lock = new ReentrantReadWriteLock();
```
[the remark this answers is not in the document]
Comment (mayankkunwar) [reply ArjunaJTA/jta/classes/com/arjuna/ats/jta/recovery/RecoveryRequired.java:7]: I will remove this class and add the AtomicBoolean to XARecoveryModule.java

File: ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest.java (hunk 1 above)
Comment (mayankkunwar) [reply ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest.java:32]: line no. 84

File: ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest2.java (hunk 3 above)
Comment (mayankkunwar) [reply ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest2.java:31]: line no. 126

File: ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryScanTest.java (hunk 4 above)
Comment (mayankkunwar) [reply ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryScanTest.java:32]: line no. 95

File: ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java [hunk 6]
```diff
@@ -78,6 +78,16 @@
 
 public class XARecoveryModule implements ExtendedRecoveryModule
 {
+	private static AtomicBoolean recoveryProblems = new AtomicBoolean(false);
```
Comment (ochaloup) [inline ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java:81]: why this needs to be `static`? I'm asking this question as I'm interested if there is some reasons behind this decision.
I assume the approach should be to not use the `static` when it's not particularly needed. The API which the behaviour exposing to outer world - aka API that was used is the call to `XARecoveryModule.isPeriodcWorkSuccesful`.
This one is then consumed by `WorkerService` (https://github.com/jbosstm/narayana/blob/5.10.4.Final/ArjunaCore/arjuna/classes/com/arjuna/ats/internal/arjuna/recovery/WorkerService.java#L114) to inform about issue on recovery process when verbose output is demanded.

File: ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java [hunk 7]
```diff
@@ -78,6 +78,16 @@
...
+
+	public static boolean isRecoveryProblems() {
```
Comment (ochaloup) [inline ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java:83]: if there is no particular reason to make anything public then it should be `private` (or package-private in case)

File: ArjunaJTA/jta/classes/com/arjuna/ats/jta/logging/jtaI18NLogger.java [hunk 8]
```diff
@@ -45,6 +45,9 @@
 /**
  * i18n log messages for the jta module.
  *
+ * For jtaI18NLogger, if any method, prefixed with <b>"warn_recovery"</b> is called then,
```
Comment (ochaloup) [inline ArjunaJTA/jta/classes/com/arjuna/ats/jta/logging/jtaI18NLogger.java:48]: We don't need this comment anymore. If you couple the functionality of the recovery problem reporting into the one place which is  the `XARecoveryModule` here then it's the responsibility of the person updating that class to consider it.
From my perspective there is no need to expose this responsibility to somebody adding a warn log message.

Comment (mayankkunwar) [reply ArjunaJTA/jta/classes/com/arjuna/ats/jta/logging/jtaI18NLogger.java:48]: okay

File: ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java (hunk 7 above)
Comment (mayankkunwar) [reply ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java:83]: I will remove public access modifier.

File: ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java (hunk 6 above)
Comment (mayankkunwar) [reply ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java:81]: Oh! sorry, I missed to see that `XARecoveryModule.isPeriodcWorkSuccesful` is already available, in order to expose the recovery state. I will remove static keyword. Thanks

File: ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (mmusgrov) [reply ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java]: The checkstyle issue I commented on was introduced as part of this PR so a separate PR would not make sense.

I admit that having to manually run the checkstyle plugin is not optimal. But I do not see an alternative to the reason I gave:

>  We have a checkstyle plugin which is skipped by default because we have a lot of old files, that we don't want to change, which would fail the compilation. But on new code we do encourage/advise that it should follow our checkstyle rules.

@ochaloup We have had this problem of  skipping the checkstyle plug on two PRs this week, it mainly affects occasional contributors. Do you have thoughts on how we could improve it (other than fixing all of the historical code).

File: ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest.java (hunk 1 above)
Comment (mmusgrov) [reply ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest.java]: Line 84 reads
```java
       } finally {
```

File: ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest2.java (hunk 3 above)
Comment (mmusgrov) [reply ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest2.java]: [Line 126](https://github.com/jbosstm/narayana/pull/1600/files#diff-ee66272224729da4319abd39b5cbd2fbR126) reads as follows:

```java
            assertFalse(!xaRecoveryModule.isPeriodicWorkSuccessful());
```

File: ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryScanTest.java (hunk 4 above)
Comment (mmusgrov) [reply ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryScanTest.java]: [Line 95](https://github.com/jbosstm/narayana/pull/1600/files#diff-24734fbaf2c75b67cc6c5de0d7795472R95) is a blank line.

File: ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (mmusgrov) [reply ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java]: I see that your latest push has fixed the checkstyle problem I reported.
But I will not mark the conversation as resolved yet since I have an outstanding question for Ondra.

File: ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryScanTest.java (hunk 4 above)
Comment (mmusgrov) [reply ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryScanTest.java]: I see that your latest push has fixed this so marking as resolved.

File: ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest.java [hunk 9]
```diff
@@ -66,8 +65,8 @@ public void testRecoveryMonitorWithFailure() throws Exception {
         recoveryEnvironmentBean.setRecoveryListener(true); // configure the RecoveryMonitor
 
         RecoveryManager manager = RecoveryManager.manager(RecoveryManager.DIRECT_MANAGEMENT);
-
-        manager.addModule(new XARecoveryModule()); // we only need to test the XARecoveryModule
```
Comment (mmusgrov) [inline ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest.java:70]: We have a policy of only making changes that are relevant to the issue being fixed. This change seems unrelated.

File: ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest2.java [hunk 10]
```diff
@@ -109,7 +108,8 @@ public void start(Xid xid, int i) throws XAException {
 
         RecoveryManager manager = RecoveryManager.manager(RecoveryManager.DIRECT_MANAGEMENT);
 
-        manager.addModule(new XARecoveryModule()); // we only need to test the XARecoveryModule
```
Comment (mmusgrov) [inline ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest2.java:112]: Same comment here (the change is unrelated to the issue)

File: ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest2.java [hunk 11]
```diff
@@ -123,7 +123,7 @@ public void start(Xid xid, int i) throws XAException {
             // check the output of the scan
             assertEquals("DONE", RecoveryMonitor.getResponse());
             assertEquals("DONE", RecoveryMonitor.getSystemOutput());
-            assertFalse(RecoveryRequired.isRecoveryProblems());
+            assertFalse(!xaRecoveryModule.isPeriodicWorkSuccessful());
```
Comment (mmusgrov) [inline ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest2.java:126]:
```java
assertTrue(xaRecoveryModule.isPeriodicWorkSuccessful());
```
 avoids the double negative.

File: ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryScanTest.java [hunk 12]
```diff
@@ -66,7 +66,8 @@ public static void beforeClass() throws Exception {
 
         manager = RecoveryManager.manager(RecoveryManager.DIRECT_MANAGEMENT);
 
-        manager.addModule(new XARecoveryModule()); // we only need to test the XARecoveryModule
```
Comment (mmusgrov) [inline ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryScanTest.java:69]: Same comment here (the change is unrelated to the issue)

File: ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest.java [hunk 13]
```diff
@@ -81,7 +80,7 @@ public void testRecoveryMonitorWithFailure() throws Exception {
             // check the output of the scan
             assertEquals("ERROR", RecoveryMonitor.getResponse());
             assertEquals("ERROR", RecoveryMonitor.getSystemOutput());
-            assertTrue(RecoveryRequired.isRecoveryProblems());
+            assertTrue(!xaRecoveryModule.isPeriodicWorkSuccessful());
```
Comment (mmusgrov) [inline ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest.java:83]:
```java
assertFalse(xaRecoveryModule.isPeriodicWorkSuccessful());
```
is easier to read

File: ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest2.java (hunk 11 above)
Comment (mmusgrov) [inline ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest2.java:126]: Same comment here

File: ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryScanTest.java [hunk 14]
```diff
@@ -92,7 +93,7 @@ public void testRecoveryMonitorWithFailure() throws Exception {
 
         manager.scan();
 
-        assertTrue(RecoveryRequired.isRecoveryProblems());
+        assertTrue(!xaRecoveryModule.isPeriodicWorkSuccessful());
```
Comment (mmusgrov) [inline ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryScanTest.java:96]: assertFalse ... is easier to read

File: ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryScanTest.java [hunk 15]
```diff
@@ -157,7 +158,7 @@ public void start(Xid xid, int i) throws XAException {
 
         manager.scan();
 
-        assertFalse(RecoveryRequired.isRecoveryProblems());
+        assertFalse(!xaRecoveryModule.isPeriodicWorkSuccessful());
```
Comment (mmusgrov) [inline ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryScanTest.java:161]: assertTrue avoids the double negative

**Comment by mmusgrov** [comment]
> Thanks @mayankkunwar! This seems good to me.
> @mmusgrov would you have some more points on this or it could be considered as ready for merge?

I had a couple comments for Mayank. And there was one for you about our checkstyle plugin.

File: ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest2.java (hunk 3 above)
Comment (mmusgrov) [reply ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest2.java]: Your latest push has resolved this so I will mark it as resolved.

Comment (mayankkunwar) [reply ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest2.java]: Now I have deleted `RecoveryRequired` class and introduced new class variable, so line 126 is using some other class xaRecoveryModule to access recovery status.

File: ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest.java (hunk 9 above)
Comment (mayankkunwar) [reply ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest.java:70]: Actually, change is related because as I have introduce new class variable in `XARecoveryModule` class to maintain recovery state, therefore in order to access the value of recovery state in test, I needed XARecoveryModule reference.

File: ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest2.java (hunk 10 above)
Comment (mayankkunwar) [reply ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest2.java:112]: same comment as above (changes are related to this issue)

File: ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest2.java (hunk 11 above)
Comment (mayankkunwar) [reply ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest2.java:126]: okay

File: ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryScanTest.java (hunk 12 above)
Comment (mayankkunwar) [reply ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryScanTest.java:69]: Changes are related

File: ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest.java (hunk 13 above)
Comment (mayankkunwar) [reply ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest.java:83]: okay

File: ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest2.java (hunk 11 above)
Comment (mayankkunwar) [reply ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest2.java:126]: okay

File: ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryScanTest.java (hunk 14 above)
Comment (mayankkunwar) [reply ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryScanTest.java:96]: okay

File: ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryScanTest.java (hunk 15 above)
Comment (mayankkunwar) [reply ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryScanTest.java:161]: okay

File: ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest.java (hunk 9 above)
Comment (mmusgrov) [reply ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest.java:70]: Ah okay, apologies for not checking

File: ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest2.java (hunk 10 above)
Comment (mmusgrov) [reply ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest2.java:112]: Thanks

Comment (mmusgrov) [reply ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryMonitorTest2.java:112]: Apologies for not checking. Thanks

File: ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryScanTest.java (hunk 12 above)
Comment (mmusgrov) [reply ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoveryScanTest.java:69]: Thanks

**Comment by mmusgrov** [comment]
Looks good. @ochaloup I will merge this one.

File: ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (ochaloup) [reply ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java]: Sorry for not reacting this sooner I wanted but I haven't got to this. I can see the discussion was moved to chat. I replied there.
